push/sender/src/main/java/org/jboss/aerogear/unifiedpush/message/holder/MessageHolderWithVariants.java

```diff
@@ -0,0 +1,71 @@
+/**
+ * JBoss, Home of Professional Open Source
+ * Copyright Red Hat, Inc., and individual contributors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.jboss.aerogear.unifiedpush.message.holder;
+
+import java.io.Serializable;
+import java.util.ArrayList;
+import java.util.Collection;
+
+import org.jboss.aerogear.unifiedpush.api.PushMessageInformation;
+import org.jboss.aerogear.unifiedpush.api.Variant;
+import org.jboss.aerogear.unifiedpush.api.VariantType;
+import org.jboss.aerogear.unifiedpush.message.UnifiedPushMessage;
+
+/**
+ * Holds push message details with what type and list of variants should be the notification sent to.
+ *
+ * Holder is used as a payload in messaging subsystem.
+ */
+public class MessageHolderWithVariants extends AbstractMessageHolder implements Serializable {
+
+    private static final long serialVersionUID = -7955411139315335655L;
+
+    private VariantType variantType;
+    private ArrayList<Variant> variants;
```

Comment:
perhaps, but I guess I wanted to have serialization under control, that's why specific type